### ISSUE TITLE
Authenticated post/put via user_token

### ISSUE THREAD

**Issue by bingneef** [buntine/discogs#49]
From the method: **make_request(path, params, method, body)** (in wrapper.rb) it seems impossible to perform authenticated non-get requests using the user token. However, I believe this is possible (and I need it for my application). Is there a specific reason you did this I am overlooking?

My proposed code would look like this:

```
# Generates a HTTP request and returns the response.
def make_request(path, params, method, body)
  full_params   = params.merge(auth_params)
  uri           = build_uri(path, full_params)
  formatted     = "#{uri.path}?#{uri.query}"
  output_format = full_params.fetch(:f, "json")
  headers       = {"Accept"          => "application/#{output_format}",
                   "Accept-Encoding" => "gzip,deflate",
                   "User-Agent"      => @app_name}

  if any_authentication?
    if [:post, :put].include?(method)
      headers["Content-Type"] = "application/json"
      if user_facing?
        @access_token.send(method, formatted, JSON(body), headers)
      else
        HTTParty.send(method, uri, {headers: headers, body: JSON(body)})
      end
    else
      if user_facing?
        @access_token.send(method, formatted, JSON(body), headers)
      else
        HTTParty.send(method, uri, headers: headers)
      end
    end
  else
    # All non-authenticated endpoints are GET.
    HTTParty.get(uri, headers: headers)
  end
end
```

I am happy to create a PR with updated specs, but checking before I put in the work :)

**Comment by buntine** [comment]
Hey there, from memory (pretty distant memory now!) there were some restrictions on the HTTP verbs allowed depending on the users authentication status.

That has quite possibly changed now. Feel free to open a pull request and I will merge it in and then do some testing before pushing to rubygems.

Thanks!

**Comment by bingneef** [comment]
Cool, thanks for the reply!

I made a fork with the code change, but won't have time to fully test it before my holiday (returning 8 aug), so will do that after.

ps. forgot to mention, thanks for the awesome Gem!

**Comment by buntine** [comment]
Hey, I've finally merged this into master and tested. I made some minor changes but the code is working correctly, I believe.

The test suite passes and I have manually tested GET,POST,PUT and DELETE as an authenticated user and also GET as an un-authenticated user.

Let me know if you wanted to run a couple of manual tests. Just pull master and try locally:

```
require "./lib/discogs"
wrapper = Discogs::Wrapper.new("Temp test", user_token: "YOUR_TOKEN")
```

Otherwise, I will push this out to Rubygems in a day or two.

cheers,
Andrew.
